### ISSUE TITLE
Version of Lasagne?

### ISSUE THREAD

**Issue by javierselva** [dbbert/dfn#7]
Hi! First of all, thanks for sharing your work!

I do not seem to be able to find the proper versions of Theano and Lasagne to make the code work. There are major changes among the different releases of these libraries that are causing me too much trouble for finding those that make the code work. Currently I am stuck with:

- Ubuntu 16.04
- Theano 0.9 
- Lasagne 0.2.dev1
- PyGPU 0.6.9 (which seems to be the proper version to work with Theano 0.9)
- CuDNN 5.1 (as more recent versions do not work properly with PyGPU 0.6.9)
- Cuda 8.0

It is the closest I feel I've been of pulling this out, however I get a Theano error where it is trying to check the type of a numpy.ndarray with ".type" which numpy arrays don't have as an attribute. 

Other combinations of Theano-Lasagne versions gave a wide range of errors where it cannot make the GPU work (related to the back-end Theano uses in more recent versions) or not being able to find specific functions (related, I guess, to some incompatibilities among specific versions of Theano and Lasagne).

It would be of great help if you could provide the used versions for making the code work. I am using Docker, so changing to different OS or Cuda/CuDNN versions wouldn't be a problem either.

Thanks in advance!

**Comment by dbbert** [comment]
Can you try with Lasagne 0.1, Theano 0.8, and Cuda 7.0? If that doesn't work you can maybe try the basic Tensorflow implementation of the bouncing mnist experiment in [experiment_bouncingMnistOriginal_tensorflow.ipynb](https://github.com/dbbert/dfn/blob/master/experiment_bouncingMnistOriginal_tensorflow.ipynb) that we added recently.

**Comment by javierselva** [comment]
Hi! Thanks for your quick response! 

Theano 0.8 and Cuda 7.0 were key to solving my problem! Finally I had to use the latest version of Lasagne given that some functions used weren't available in the 0.1 release. Also, I had some other issues that I finally managed to solve (regarding the .theanorc configuration file). 

I left the [Dockerfile I used to run the code here](https://drive.google.com/drive/folders/1WJg_I6WBHTnKG55VsphuUdESiOB49jTH?usp=sharing) in case somebody else may need it.

Thanks again!
